vector_setters.go

```diff
@@ -175,5 +374,74 @@ func (vec *vector) setMap(rowIdx C.idx_t, val any) {
 		i++
 	}
 
-	vec.setList(rowIdx, list)
+	return setList(vec, rowIdx, list)
+}
+
+func setUUID[S any](vec *vector, rowIdx C.idx_t, val S) error {
+	var uuid UUID
+	switch v := any(val).(type) {
+	case UUID:
+		uuid = v
+	default:
+		return castError(reflect.TypeOf(val).String(), reflect.TypeOf(uuid).String())
+	}
+	hi := uuidToHugeInt(uuid)
+	setPrimitive(vec, rowIdx, hi)
+	return nil
+}
+
+// TODO: Support ARRAY UNION BIT ANY VARINT,
+func setVectorVal[S any](vec *vector, rowIdx C.idx_t, val S) error {
```

Comment:
I don't see any tests for the switch here, but I realize that will be tricky without the respective data chunk. We can keep this, but let's later add a test for each type in the table UDFs. Similar to `func TestAllTypesScalarUDF(t *testing.T)`.

Reply:
Yes, I agree that would be good, will resolve this for now